server/ai_process.go

```diff
@@ -1013,41 +1013,25 @@ func submitLiveVideoToVideo(ctx context.Context, params aiRequestParams, sess *A
 		}
 		return nil, err
 	}
+	// Send request to orchestrator
 	resp, err := client.GenLiveVideoToVideoWithResponse(ctx, req)
 	if err != nil {
 		return nil, err
 	}
 	if resp.JSON200 != nil {
 		// append orch hostname to the given url if necessary
-		appendHostname := func(urlPath string) (*url.URL, error) {
-			if urlPath == "" {
-				return nil, fmt.Errorf("invalid url from orch")
-			}
-			pu, err := url.Parse(urlPath)
-			if err != nil {
-				return nil, err
-			}
-			if pu.Hostname() != "" {
-				// url has a hostname already so use it
-				return pu, nil
-			}
-			// no hostname, so append one
-			u := sess.Transcoder() + urlPath
-			return url.Parse(u)
-		}
-		pub, err := appendHostname(resp.JSON200.PublishUrl)
+		pub, err := url.Parse(resp.JSON200.PublishUrl)
 		if err != nil {
-			return nil, fmt.Errorf("pub url - %w", err)
+			return nil, fmt.Errorf("invalid publish URL: %w", err)
 		}
-		sub, err := appendHostname(resp.JSON200.SubscribeUrl)
+		sub, err := url.Parse(resp.JSON200.SubscribeUrl)
```

Comment:
as mentioned earlier - preference for not changing this. this will still work even if we have a hostname.

Reply:
Moved the hostname logic so that it uses the original value from `sess.Transcoder()` on the gateway, accurately identifying the remote IP of the orchestrator. It's appended on the orchestrator side of the request processing before being sent to the ai-runner